TF_Service_dotNet/TouchFree/Connections/MessageQueues/MessageQueueHandler.cs

```diff
@@ -41,55 +40,39 @@ protected virtual void OnUpdate()
 
         private void CheckQueue()
         {
-            IncomingRequest content;
-            if (queue.TryPeek(out content))
-            {
-                // Parse newly received messages
-                queue.TryDequeue(out content);
-                if (ActionCodes.Contains(content.action))
-                {
-                    ValidateRequestIdAndHandleRequest(content);
-                }
-                else
-                {
-                    TouchFreeLog.ErrorWriteLine($"Unexpected ActionType of {content.action} in {GetType().Name}");
-                }
-            }
-        }
-
-        private void ValidateRequestIdAndHandleRequest(IncomingRequest _request)
-        {
-            JObject contentObj = JsonConvert.DeserializeObject<JObject>(_request.content);
-
-            if (!RequestIdExists(contentObj))
-            {
-                CreateAndSendNoRequestIdError(_request);
-            }
-            else
-            {
-                Handle(_request, contentObj, contentObj.GetValue("requestID").ToString());
-            }
+            if (queue.IsEmpty) return;
+            
+            // Parse newly received messages
+            if (!queue.TryDequeue(out var content)) return;
+            HandleRequest(content);
         }
 
-        protected virtual void CreateAndSendNoRequestIdError(IncomingRequest _request)
-        {
-            ResponseToClient failureResponse = new ResponseToClient(string.Empty, "Failure", noRequestIdFailureMessage, _request.content);
+        private void HandleRequest(IncomingRequest request) =>
+            request.DeserializeAndValidateRequest(jObject => ValidateContent(jObject, request))
+                .Match(Handle, error => HandleValidationError(request, error));
```

Comment:
With this version of error handling we never send back error messages with requestIDs set and so it will not trigger callbacks in tooling for those messages